pkg/scheduler/framework/plugins/clustereviction/cluster_eviction.go

```diff
@@ -47,11 +48,30 @@ func (p *ClusterEviction) Name() string {
 }
 
 // Filter checks if the target cluster is in the GracefulEvictionTasks which means it is in the process of eviction.
-func (p *ClusterEviction) Filter(_ context.Context, bindingSpec *workv1alpha2.ResourceBindingSpec, _ *workv1alpha2.ResourceBindingStatus, cluster *clusterv1alpha1.Cluster) *framework.Result {
+func (p *ClusterEviction) Filter(_ context.Context, bindingSpec *workv1alpha2.ResourceBindingSpec, bindingStatus *workv1alpha2.ResourceBindingStatus, cluster *clusterv1alpha1.Cluster) *framework.Result {
+	failoverHistory := bindingStatus.FailoverHistory
+	if len(failoverHistory) != 0 {
+		lastFailover := failoverHistory[len(failoverHistory)-1]
+		if containsCluster(lastFailover.ClustersBeforeFailover, cluster.Name) {
+			klog.V(2).Infof("Workload has been failed over from this cluster %s.", cluster.Name)
+			return framework.NewResult(framework.Unschedulable, fmt.Sprintf("workload has been failed over from this cluster %s", cluster.Name))
+		}
```

Comment:
I know in your scenario an application can only failover from one cluster to another. 
Even though I'm not sure if there is a case that an application is divided into multiple clusters and failover together yet, I think it would be great to support that case here.
In other words, here `failover from` might be not just one cluster. 

See https://github.com/karmada-io/karmada/pull/5251#issuecomment-2439277014.

Reply:
Perhaps the log line just needs updating. The clustersBeforeFailover already covers scenarios in which replicas are divided across different clusters. Since clustersBeforeFailure only include clusters that have been marked for eviction, we shouldn’t reschedule replicas to those clusters. 

That’s why I added this filter step

Reply:
> Since clustersBeforeFailure only include clusters that have been marked for eviction, we shouldn’t reschedule replicas to those clusters.

Maybe I have a different view on this. The `clusterBeforeFailover` is designed to hold all clusters before the failover, including the normal clusters and those targeted to be evicted.

This piece of code will filter all clusters out, not just the one marked for eviction, right?
```golang
if containsCluster(lastFailover.ClustersBeforeFailover, cluster.Name) {
	klog.V(2).Infof("Workload has been failed over from this cluster %s.", cluster.Name)
	return framework.NewResult(framework.Unschedulable, fmt.Sprintf("workload has been failed over from this cluster %s", cluster.Name))
}
```

Reply:
We added the UpdateFailoverStatus method within the `evictBinding` method of the rb application failover controller. Based off https://github.com/karmada-io/karmada/blob/master/pkg/controllers/applicationfailover/rb_application_failover_controller.go#L135, the only clusters passed into the eviction method are clusters for which replicas are unhealthy, and where the toleration time has elapsed. 

So we'll only be appending clusters that are meant for eviction to the `clustersBeforeFailover`, which is why we want to filter them out before scheduling.

Reply:
I guess my understanding of the `clustersBeforeFailover` attribute was that we're tracking only the clusters that are undergoing eviction - that way we can track which replicas have been failed over (from their originating clusters to their destinations).